vtex/faststore
# Feat: PLP multiple page template

Depends on: this code will change after:

- https://github.com/vtex/faststore/pull/2215
- https://github.com/vtex/vtex-cicd-platform/pull/572
- https://github.com/vtex/faststore/pull/2227

## What's the purpose of this pull request?

This PR aims to find the best PLP template from the CMS, based on the PLP slug.

`vendemo` account
<img width="1418" alt="Screenshot 2024-02-06 at 18 44 14" src="https://github.com/vtex/faststore/assets/11325562/12ac9efe-af47-4ac3-9432-6529b0c1bf2d">

## How it works?

It iterates the slug until there are no slashes (/), prioritizing the following order:
 
1. A PLP template that matches the subcategory (e.g. slug = /department/category/subcategory).
2. A PLP template that matches the category (e.g. slug = /department/category).
3. A PLP template that matches the department (e.g. slug = /department).
4. If no matches are found, use the generic PLP template.

## How to test it?

1. localhost Initial setup:
You can change the tenant from CMS to the account `vendemo`. This store has all the PLPs templates created and published. Also, we changed the stored to `lojavivara`, for the sake of testing with a more complete catalog without harming the devs using `storeframework` nowadays.

<img width="924" alt="Screenshot 2024-02-23 at 13 26 42" src="https://github.com/vtex/faststore/assets/11325562/3a3a3654-459d-4761-8b05-50401495bd12">

<img width="796" alt="Screenshot 2024-02-23 at 13 26 02" src="https://github.com/vtex/faststore/assets/11325562/d67c89db-38e6-419d-9fa3-f900ff9dec11">

2. Note that we have different templates created for PLPs with all scenarios mentioned above. So, use the following PLPs to test


|Matching Type|URL|
|-|-|
|Slug Subcategory - /vivara/joias/aneis |http://localhost:3000/vivara/joias/aneis |
|Slug  Category - /vivara/joias |http://localhost:3000/vivara/joias |
|Slug  Department - /vivara |http://localhost:3000/vivara |
|General - /life |http://localhost:3000/life |

### Starters Deploy Preview

This PR in starter uses the same configs mentioned. `vendemo` for CMS and `lojavivara` for API:
- https://github.com/vtex-sites/starter.store/pull/389

You can test using the same URLs:

|Matching Type|URL|
|-|-|
|Slug Subcategory - /vivara/joias/aneis |https://sfj-047d489--starter.preview.vtex.app/vivara/joias/aneis |
|Slug  Category - /vivara/joias |https://sfj-047d489--starter.preview.vtex.app/vivara/joias |
|Slug  Department - /vivara |https://sfj-047d489--starter.preview.vtex.app/vivara |
|General - /life |https://sfj-047d489--starter.preview.vtex.app/life |


## References

Doc PR
- https://github.com/vtex/faststore/pull/2236
